bgpd/bgp_routemap.c

```diff
@@ -1097,7 +1097,7 @@ route_match_vrl_source_vrf(void *rule, const struct prefix *prefix,
 	if (strncmp(vrf_name, "n/a", VRF_NAMSIZ) == 0)
 		return RMAP_NOMATCH;
 
-	if (path->extra == NULL)
+	if (path->extra == NULL || path->extra->bgp_orig == NULL)
```

Comment:
I believe OR should be `(path->extra && path->extra->bgp_orig == NULL)`

Reply:
Yes, that is correct, should be changed to AND.

Reply:
I think OR should be fine here, 
else we would dereference NULL in Line:1103

Reply:
No change.